pcdscalc/be_lens_calcs.py

```diff
@@ -0,0 +1,830 @@
+                         'please provide a number from 1 to %s ',
+                         set_number_top_to_bot, len(sets))
+            return
+        return sets[set_number_top_to_bot - 1]
+
+
+def set_lens_set_to_file(sets_list_of_tuples, filename, make_backup=True):
+    '''
+    # The Be lens holders can take 3 different sets that we usually set before
+    # few experiments so we only vent the relevant beamline section once.
+    # We then store these sets into a config file that we can call from some
+    # methods. Later, we save this config file to the specific
+    # experiment so users can make sure that they know which stack was used
+    # for there beamtime.
+
+    Usage:
+
+    .. code-block:: python
+
+        sets_list_of_tuples = [(3, 0.0001, 1, 0.0002),
+                               (1, 0.0001, 1, 0.0003, 1, 0.0005),
+                               (2, 0.0001, 1, 0.0005)]
+        set_lens_set_to_file(sets_list_of_tuples, ../path/to/lens_set)
+
+    Parameters
+    ----------
+    sets_list_of_tuples : `list`
+        List with tuples for lens sets
+    filename : `str`
+        Path to the filename to set the lens sets list to
+    make_backup : `bool`
+        To indicate if a backup file should be created or not.
+        Default = True
+    '''
+    # Make a backup with today's date
+    if make_backup:
+        backup_path = filename + str(date.today()) + '.bak'
+        try:
+            shutil.copyfile(filename, backup_path)
+        except Exception as ex:
+            logger.error('Something went wrong with copying the file %s', ex)
+            pass
+    with open(filename, 'w') as lens_file:
+        # TODO: do we want to pprint.pformat() like in the old code?
+        # lens_file.write(pprint.pformat(sets_list_of_tuples))
+        lens_file.write(sets_list_of_tuples)
+
+
+def get_att_len(energy, material="Be", density=None):
+    '''
+    Get the attenuation length (in meter) of a material
+
+    The X-ray beam intensity I(x) at depth x in a material is a function
+    of the attenuation coefficient mu, and can be calculated by the
+    Beer-Lambert law.
+    The Absorption Length (or Attenuation Length) is defined as the distance
+    into a material where the x-ray beam intensity has decreased to a value
+    of 1/e (~ 40%) of the incident beam intensity (Io)
+
+    (1/e) = e^(-mu * x)
+    ln(1/e) = ln(e^(-mu * x))
+    1 = mu * x
+    x = 1/mu
+
+    Usage:
+
+    .. code-block:: python
+
+        get_att_len(energy=8, material='Be')
+
+    Parameters
+    ----------
+    energy : number
+        Beam Energy given in KeV
+    material : `str`
+        Default - Beryllium.
+        The use of beryllium extends the range of operation
+        of compound refractive lenses, improving transmission,
+        aperture size, and gain
+    density : `float`
+        Material density in g/cm^3
+
+    Returns
+    -------
+    att_len : `numpy.float64`
```

Comment:
```suggestion
    att_len : float
```

No need to be specific about this numpy type here - it functions just as a regular Python float.